### ISSUE TITLE
[sanity_client] Cannot perform queries

### ISSUE THREAD

**Issue by muccy** [vyuh-tech/vyuh#9]
I'm using `sanity_client: ^1.2.1` but I cannot complete a basic query like:
```groq
*[_type == "event"]{
  name,
  date,
  venue->{
    name
  },
  headline->{
    name
  },
  image,
  tickets
}
```

The reported error is:
```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: type 'List<dynamic>' is not a subtype of type 'Map<String, dynamic>?' in type cast
#0      _$ServerResponseFromJson (package:sanity_client/response_types.g.dart:17:30)
#1      new ServerResponse.fromJson (package:sanity_client/response_types.dart:51:7)
#2      SanityClient._getQueryResult (package:sanity_client/client.dart:145:47)
#3      SanityClient.fetch (package:sanity_client/client.dart:111:12)
<asynchronous suspension>
#4      _HomeState.build.<anonymous closure> (package:flutter_sanity_test/src/home.dart:57:34)
```

Looking at the JSON Serializable generated code I see:
```dart
ServerResponse _$ServerResponseFromJson(Map<String, dynamic> json) =>
    ServerResponse(
      result: json['result'] as Map<String, dynamic>?,
      ms: json['ms'] as int,
      query: json['query'] as String,
    );
```

But `json["result"]` isn't a Map, but a List.

I could think about a fix, but I don't know if you set Map type for a reason (I'm totally unexperienced with Sanity.io and GROQ)

**Comment by pavanpodila** [comment]
Yup looks like a bug in the client where the type has been set to a Map instead of dynamic.

**Comment by pavanpodila** [comment]
This is now fixed in the latest version: ^1.2.2, with a test as well. Please update and try again @muccy. I'll close it once you verify.

**Comment by pavanpodila** [comment]
Closing it for now ... pls reopen if this is still an issue
